### PR TITLE
Unify names in Utils

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Unify the mean utils aka one was `exceptions` other `warning_utils`

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: pytorch_lightning/utilities/xla_device_utils.py [hunk 1]
```diff
@@ -1,104 +1,6 @@
...
-# limitations under the License.
-import functools
-import importlib
-import queue as q
-import traceback
-from multiprocessing import Process, Queue
+from warnings import warn
 
-import torch
+warn("`xla_device_utils` package has been renamed to `xla_device` since v1.2 and will be removed in v1.3",
```
Comment (rohitgr7) [inline pytorch_lightning/utilities/xla_device_utils.py:3]: deprecation and removal should be 0.2 apart, right? 1.2 and 1.4?

File: pytorch_lightning/overrides/data_parallel.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from torch.nn.parallel._functions import Gather
 
 from pytorch_lightning.core.step_result import Result
-from pytorch_lightning.utilities.warning_utils import WarningCache
+from pytorch_lightning.utilities.warnings import WarningCache
```
Comment (rohitgr7) [inline pytorch_lightning/overrides/data_parallel.py:27]: should the name for the module be different from stdlib modules? https://github.com/PyTorchLightning/pytorch-lightning/issues/4436

Comment (Borda) [reply pytorch_lightning/overrides/data_parallel.py:27]: well, do you have a suggestion for a better name? :]

File: pytorch_lightning/utilities/xla_device_utils.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/utilities/xla_device_utils.py]: fixed

File: pytorch_lightning/overrides/data_parallel.py (hunk 2 above)
Comment (rohitgr7) [reply pytorch_lightning/overrides/data_parallel.py:27]: pl_warnings and pl_argparse maybe :sweat_smile:

Comment (Borda) [reply pytorch_lightning/overrides/data_parallel.py:27]: in such case, we would need to. add pl to everything :D

**Review by awaelchli — APPROVED** [review]
did you change something in the files? I think they should show up as renamed/moved without a large diff display

**Comment by Borda** [comment]
> did you change something in the files? I think they should show up as renamed/moved without a large diff display

no, the content is untouched...
